src/main/scala/org/allenai/plugins/DeployPlugin.scala

```diff
@@ -348,10 +540,22 @@ object DeployPlugin extends AutoPlugin {
       System.console.readLine()
       ConfigFactory.empty
     }
-    // Write the fully-resolved environment configuration to disk.
-    val targetFile = targetDir / URLEncoder.encode(hostName, "utf-8") / "env.conf"
-    log.info(s"Writing environment config for '$deployEnv' to '${targetFile.getPath}'")
-    IO.write(targetFile, fullEnvConf.root.render(ConfigRenderOptions.concise))
+
+    // Resolve and write config files for all replicas, including their specified overrides.
+    deployConfig.replicasByHost foreach {
+      case (host, replicas) => replicas foreach { replica =>
+        // Encode hostname to reduce chances of naming problems.
```

Comment:
This is a *tiny* bit worrying. Does it retain `.` literals, or are those encoded? If they are encoded, I'd almost prefer this just fail when you get a weird hostname.